### PR TITLE
Doc for adding docs

### PR DESCRIPTION
**Pull request by plauche**

### CODE REVIEW

**Comment by plauche** [comment]
Ready for initial review. I think this covers most of our bases. Let me know if I've missed any crucial or helpful topics.

**Comment by plauche** [comment]
[Add a doc for contributing to the docs](https://trello.com/c/TEP31d85/920-add-a-doc-for-contributing-to-the-docs)

File: docs/contributing/documentation.rst [hunk 1]
```diff
@@ -1,51 +1,119 @@
...
+    - Verify that any new hyperlinks work as intended
+    - Make sure that ``tools/gendocs.py`` runs successfully without throwing any errors or warnings. Fix all warnings until the script runs cleanly.
+
+General Docs
+------------
+
+All docs files under ``docs\`` are written in the `reStructuredText <http://docutils.sourceforge.net/rst.html>`__ format.
+Please refer to the :doc:`standards doc <kubos-standards>` for more specific writing standards.
+
+All executable projects, such as clients, examples or services, should have a ``README.md`` which explains the purpose of the project,
```
Comment (cantino) [inline docs/contributing/documentation.rst]: *examples,*

File: docs/contributing/documentation.rst [hunk 2]
```diff
@@ -1,51 +1,119 @@
...
+
+- doxygen
+- graphvis
+- plantuml
+- python 2.7 + pip
+
+Python dependencies (most likely installed with ``pip``):
+
+- Sphinx v1.5.6
+- breathe
```
Comment (jacoffey3) [inline docs/contributing/documentation.rst]:
```suggestion
- breathe v4.11.1
```

File: docs/conf.py [hunk 3]
```diff
@@ -18,9 +18,17 @@
 #
 import os
 import sys
-sys.path.insert(0, os.path.abspath('../apis/pumpkin-mcu-api'))
-sys.path.insert(0, os.path.abspath('../apis/python-app-api'))
-sys.path.insert(0, os.path.abspath('../hal/python-hal/i2c'))
+
+# List of paths to Kubos python module
+# These are injected into the system path so Sphinx can find them
```
Comment (CatGarab) [inline docs/conf.py]: :fire:

File: docs/contributing/documentation.rst [hunk 4]
```diff
@@ -1,51 +1,119 @@
...
+To verify your docs:
+
+    - Make sure that any new ``*.rst`` files are accessible through normal page clicks if you start at the top-level ``index.html``
+    - Verify that any new hyperlinks work as intended
+    - Make sure that ``tools/gendocs.py`` runs successfully without throwing any errors or warnings. Fix all warnings until the script runs cleanly.
+
+General Docs
+------------
+
+All docs files under ``docs\`` are written in the `reStructuredText <http://docutils.sourceforge.net/rst.html>`__ format.
```
Comment (CatGarab) [inline docs/contributing/documentation.rst]: nit: `docs\` -> the `docs/` folder

File: docs/contributing/documentation.rst [hunk 5]
```diff
@@ -1,51 +1,119 @@
...
+
+All executable projects, such as clients, examples or services, should have a ``README.md`` which explains the purpose of the project,
+basic usage, and any runtime configuration options.
+
+Documenting Rust
+----------------
+
+The Rust book provides a pretty good overview of the standards for documenting Rust code `here <https://doc.rust-lang.org/1.24.0/book/first-edition/documentation.html>`__.
+
+We leverage the ``cargo doc`` tool in generating our Rust documentation, so all Rust source is expected to confirm
```
Comment (CatGarab) [inline docs/contributing/documentation.rst]: confirm -> conform

File: docs/contributing/documentation.rst [hunk 6]
```diff
@@ -1,51 +1,119 @@
...
-Doxygen
-~~~~~~~
+The Python source in the Kubos repo follows the `PEP 257 <https://www.python.org/dev/peps/pep-0257/>`__ docstring conventions. 
+
+Public facing functions, classes, and modules should all be documented with the appropriate, descriptive docstrings.
+
+Including Python Docs
+~~~~~~~~~~~~~~~~~~~~~
+
+Any new python modules will need to be added to the ``python_modules`` list in``docs/conf.py`` file
```
Comment (CatGarab) [inline docs/contributing/documentation.rst]: nit: space after "in"

Comment (CatGarab) [inline docs/contributing/documentation.rst]: Might be worth having "docs/conf.py" link to the actual file in our repo?

File: docs/contributing/documentation.rst [hunk 7]
```diff
@@ -1,51 +1,119 @@
...
+----------------
+
+The Rust book provides a pretty good overview of the standards for documenting Rust code `here <https://doc.rust-lang.org/1.24.0/book/first-edition/documentation.html>`__.
+
+We leverage the ``cargo doc`` tool in generating our Rust documentation, so all Rust source is expected to confirm
+to the standards laid out in the book.
+
+Documentation for any individual crate can be generated by running ``cargo doc`` from within that crate's folder.
+The generated docs can be found in ``kubos/target/doc/<crate-name>``.
+
```
Comment (CatGarab) [inline docs/contributing/documentation.rst]: Need a blurb about linking to new Rust docs. Our `gendocs.py` script puts them all in the dedicated "html/rust-docs" folder and then you have to link to them like this:
```
.. |comms-service| raw:: html

    <a href="../../rust-docs/comms_service/index.html" target="_blank">Framework Rust documentation</a>
```

File: docs/contributing/documentation.rst [hunk 8]
```diff
@@ -1,51 +1,132 @@
...
+We leverage the ``cargo doc`` tool in generating our Rust documentation, so all Rust source is expected to conform
+to the standards laid out in the book.
+
+Documentation for any individual crate can be generated by running ``cargo doc`` from within that crate's folder.
+The generated docs can be found in ``kubos/target/doc/<crate-name>``.
+
+Including Rust Docs
+~~~~~~~~~~~~~~~~~~~
+
+The rust docs generated by ``cargo doc`` will be placed in ``kubos/target/doc/<crate-name>``. These files are not accessible to the rest of the docs, so the contents of ``kubos/target/doc`` are moved to ``html/rust_docs`` during the documentation generation process.
```
Comment (CatGarab) [inline docs/contributing/documentation.rst]: nit: capitalize Rust (also in the next line)

File: docs/contributing/documentation.rst [hunk 9]
```diff
@@ -1,51 +1,132 @@
...
+
+Generating The Docs
+-------------------
+
+The Kubos docs are generated using `Sphinx <http://www.sphinx-doc.org/en/master/>`__ in conjunction with a variety of tools including
+`rustdoc <https://doc.rust-lang.org/rustdoc/what-is-rustdoc.html>`__ and `Doxygen <http://www.doxygen.nl/>`__.
+We suggest generating the docs from inside of the Kubos SDK environment, as it already has all dependencies installed.
+
+If you would like to generate the docs outside of the SDK environment, please refer to the following 
+list of dependencies, in addition to the list of `build dependencies<build-dependencies>`__.
```
Comment (CatGarab) [inline docs/contributing/documentation.rst]: The link isn't being generated correctly. Need a space before the `<`

File: docs/contributing/documentation.rst [hunk 10]
```diff
@@ -1,51 +1,132 @@
...
+
+Generating The Docs
+-------------------
+
+The Kubos docs are generated using `Sphinx <http://www.sphinx-doc.org/en/master/>`__ in conjunction with a variety of tools including
+`rustdoc <https://doc.rust-lang.org/rustdoc/what-is-rustdoc.html>`__ and `Doxygen <http://www.doxygen.nl/>`__.
+We suggest generating the docs from inside of the Kubos SDK environment, as it already has all dependencies installed.
+
+If you would like to generate the docs outside of the SDK environment, please refer to the following 
+list of dependencies, in addition to the list of `build dependencies <build-dependencies>`__.
```
Comment (CatGarab) [inline docs/contributing/documentation.rst]: whoops, I was wrong about why this link is broken.

1. I think it's supposed to be a reference
```
:ref:`build dependencies <build-dependencies>`
```
2. But I couldn't find a reference or doc that matched

Comment (plauche) [reply docs/contributing/documentation.rst]: Whoops, that was a link to a section not yet created. I went ahead and added a stubbed reference under Getting Started.

**Pull request merged**